### PR TITLE
fix: resolving rimraf bin dir

### PR DESCRIPTION
**Pull request by tada5hi**
## Description
Due to the fact that rimraf has changed its binary path, lerna no longer works with the new version 4.4.x. Therefore the cli location is now read from the package.json, as fallback the original module resolving was used.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (change that has absolutely no effect on users)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

fixes [#3614](https://github.com/lerna/lerna/issues/3611)

### CODE REVIEW

File: libs/core/src/lib/rimraf-dir.ts [hunk 1]
```diff
@@ -1,36 +1,59 @@
...
 
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const childProcess = require("@lerna/child-process");
 
-// NOTE: if rimraf moves the location of its executable, this will need to be updated
-const RIMRAF_CLI = require.resolve("rimraf/bin");
+let rimrafBinPath: string | undefined;
 
-export function rimrafDir(dirPath: string) {
+export async function useRimrafBinPath(): Promise<string> {
```
Comment (JamesHenry) [inline libs/core/src/lib/rimraf-dir.ts:13]: Please can we not export this to make it clear that it is not used outside this file

```suggestion
async function useRimrafBinPath(): Promise<string> {
```

**Comment by JamesHenry** [comment]
Thanks @tada5hi one small change then this LGTM!

File: libs/core/src/lib/rimraf-dir.ts (hunk 1 above)
Comment (tada5hi) [reply libs/core/src/lib/rimraf-dir.ts:13]: good point ^^, sry :see_no_evil: 
I will fix that asap

Comment (JamesHenry) [reply libs/core/src/lib/rimraf-dir.ts]: You can commit the suggestion directly in the github UI just FYI

Comment (tada5hi) [reply libs/core/src/lib/rimraf-dir.ts]: oh, wasn't aware of that ^^. Now i have already made a commit from my fork, but good to know, thanks

**Review by JamesHenry — APPROVED** [review]

**Comment by JamesHenry** [comment]
Thanks again!

**Comment by JamesHenry** [comment]
Sadly this didn't actually solve the problem because the bin path is not part of the package's exports and so node blocks it, I am looking into refactoring the whole approach here, I believe we can just use the node API of rimraf instead of all these child process + bin shenanigans

**Comment by tada5hi** [comment]
damm ... i think the problem is that i missed to join the `pck.bin` path with the directory path of require.resolve.

**Pull request merged**